src/charm.py

```diff
@@ -47,6 +53,16 @@ def __init__(self, *args: typing.Any) -> None:
             strip_prefix=True,
         )
 
+    def _update_status(self, status: StatusBase) -> None:
+        """Update the application and unit status.
+
+        Args:
+            status: the desired application and unit status.
+        """
+        self.unit.status = status
+        if self.unit.is_leader():
+            self.app.status = status
```

Comment:
This is just a helper function to set both the unit status and app status to the same thing and it's not suited for other situation. I updated the function name to `_update_app_and_unit_status` to avoid confusions. Thanks!

Reply:
Well it doesn't simply set the app and unit status.
I sets the unit status and if it's the leader sets the app status to same status of the leader unit

So if I get it correctly if a unit is failing but the leader is ok, the app status will be Active even if a unit has a problem.
I think we should have something a bit smarter for the app status